libs/component-tests/src/main/java/org/codice/ddf/test/component/rules/TestFailureLogger.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.test.component.rules;
+
+import java.util.Arrays;
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableMap;
+
+public class TestFailureLogger extends TestWatcher {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TestFailureLogger.class);
+
+    private static final String BUNDLE_LOG_FORMAT = "%3s | %12s | %20s | %s";
+
+    private static final Map<Integer, String> BUNDLE_STATES =
+            new ImmutableMap.Builder<Integer, String>().put(Bundle.UNINSTALLED, "UNINSTALLED")
+                    .put(Bundle.INSTALLED, "INSTALLED")
+                    .put(Bundle.RESOLVED, "RESOLVED")
+                    .put(Bundle.STARTING, "STARTING")
+                    .put(Bundle.STOPPING, "STOPPING")
+                    .put(Bundle.ACTIVE, "ACTIVE")
+                    .build();
+
+    @Override
+    protected void failed(Throwable e, Description description) {
```

Comment:
❗️ It looks like this doesn't log the stacktrace anywhere. Can we do something like https://github.com/codice/ddf/blob/4cb34066fcbd9cf9a885c41c6336374dc4e8f0f0/distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/utils/LoggingUtils.java#L29 instead?